src/storage.rs

```diff
@@ -59,11 +81,29 @@ pub struct AccountBasic {
 }
 
 impl AccountBasic {
+    /// Create a new account with the given balance, nonce, code hash and code.
+    pub fn new(balance: U256, nonce: u64, code_hash: B256, code: EvmCode) -> Self {
+        AccountBasic {
+            balance,
+            nonce,
+            code: Some(code),
+            code_hash: Some(code_hash),
+        }
+    }
+
     /// Check if an account is empty for removal per EIP-161
     // https://github.com/ethereum/EIPs/blob/96523ef4d76ca440f73f0403ddb5c9cb3b24dcae/EIPS/eip-161.md
     pub fn is_empty(&self) -> bool {
         self.balance == U256::ZERO && self.nonce == 0 && self.code.is_none()
     }
+
+    /// Create a new account with the given balance.
+    pub fn from_balance(balance: U256) -> Self {
```

Comment:
It looks like all use cases combine this with `impl From<AccountBasic> for EvmAccount`. 

I wonder if we can simplify both here and at call sites by providing `EvmAccount::with_balance` instead?